src/resolve/metadataResolver.ts

```diff
@@ -31,6 +31,20 @@ import { NodeFSTreeContainer, TreeContainer } from './treeContainers';
 Messages.importMessagesDirectory(__dirname);
 const messages = Messages.loadMessages('@salesforce/source-deploy-retrieve', 'sdr');
 
+/**
+ * Checks if the given path is a web_app bundle directory.
+ * web_app bundles don't have metadata XML files and are identified by directory structure.
+ */
+const isWebAppBundlePath = (path: string): boolean => {
```

Comment:
can we export the `isWebAppBaseType` method from `digitalEXperienceSourceADapter` to keep it DRY